### PR TITLE
Adding v3 folder to begin Private IP addressing

### PR DESCRIPTION
**Pull request by matthewdfuller**
Exact same files except all instances of `COREOS_PUBLIC_IPV4` have been replaced with `COREOS_PRIVATE_IPV4` and all instances of `public-hostname` have been replaced with `local-hostname` for the AWS metadata service.

### CODE REVIEW

**Comment by chr0n1x** [comment]
hahaha oh boy

File: v3/fleet/docker-cleanup.service [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+After=docker.service
+Requires=docker.service
+
+[Service]
+User=core
+Restart=on-failure
+RestartSec=20
+TimeoutStartSec=0
+ExecStartPre=/usr/bin/systemctl is-active bootstrap
+ExecStart=/usr/bin/sudo bash /home/core/mesos-systemd/v2/util/docker-cleanup.sh
```
Comment (chr0n1x) [inline v3/fleet/docker-cleanup.service]: lol so uh........ notice the `v2`

File: init [hunk 2]
```diff
@@ -2,17 +2,25 @@
 
 echo "-------Beginning bootstrap script-------"
 
+if [ -z "$1" ]; then
+  VERSION="v2"
+else
+  VERSION="$1"
```
Comment (chr0n1x) [inline init:8]: might want to store the version in etcd instead? I don't know anymore

File: v3/fleet/docker-cleanup.service (hunk 1 above)
Comment (matthewdfuller) [inline v3/fleet/docker-cleanup.service]: thought I had them all :( i'll update in a second

File: v3/util-units/create-users.service [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+EnvironmentFile=/etc/environment
+User=core
+Type=oneshot
+RemainAfterExit=false
+
+ExecStartPre=-/usr/bin/rm -rf /home/core/mesos-users
+
+# TODO: re-visit this - dir should probably be configurable
+# look at the script to see what it's doing - you just need a repo with user public keys
+ExecStart=/usr/bin/bash -c '/usr/bin/git clone git@github.com:behance/mesos-users /home/core/mesos-users && /home/core/mesos-systemd/v2/util/add_users.sh /home/core/mesos-users/users'
```
Comment (chr0n1x) [inline v3/util-units/create-users.service]: boop

File: init (hunk 2 above)
Comment (matthewdfuller) [inline init:8]: I did it this way so we can customize the bootstrap command in the CF params to run v3 if needed. But etcd could work too

Comment (chr0n1x) [inline init:8]: well if it's in etcd any scripts that reference util scripts can utilize it too

**Comment by evanwieren** [comment]
:+1:

**Pull request merged**